### PR TITLE
Refactored PointerLockControls

### LINKED ISSUE #16160: PointerLockControls needs better implementation
##### PointerLockControls needs better implementation consistent and compatible with other controls.

While trying to validate PR #16093 I noticed that `PointerLockControls` does not play nice with OrbitControls and implements several anti-patterns that result with strange and unexpected behaviors.

Some of the problems:

1. It sets camera rotation to 0,0,0 in the constructor (because otherwise it does not work correctly).

2. It moves camera from its original hierarchy to its own Object3D hierarchy.

3. It requires its `yawObject` and its children to be added to rendered scene (otherwise transforms are not applied to camera)

4. Unlike other controls, It does not emit `change` event.

5. In the example (misc_controls_pointerlock.html), there is an application-side implementation of first person controls in conjunction with `PointerLockControls` that exhibits convoluted patterns that require understanding on internal hierarchy of `PointerLockControls` and its (internally set) relationship with the camera.

Suggested solution:

Re-implement `PointerLockControls` in such way that it does not reset camara's rotation, it does not make any modifications to any scenegraph, and can be intuitively used like other controls.

### PR DESCRIPTION
**Pull request by arodic**
This fixes #16160 

With this new implementation, `PointerLockControls` does not change camera hierarchy and it does not require camera rotation to be 0,0,0.

In addition, this implementation exposes `reverseY`, `sensitivity`, `minPolarAngle` and `maxPolarAngle` similar to `OrbitControls`, and it emits `change` event like other controls.

This is a breaking change because it requires some code to be removed from applications using it. Most notably, there is no need to do `scene.add( controls.getObject() );`.

### CODE REVIEW

File: examples/js/controls/PointerLockControls.js [hunk 1]
```diff
@@ -83,28 +116,33 @@ THREE.PointerLockControls = function ( camera, domElement ) {
...
 
-		return yawObject;
+		console.warn( 'THREE.PointerLockControls: getObject() has been deprecated.' );
+
+		return scope.object;
 
 	};
 
-	this.getDirection = function () {
+	this.getDirection = function ( v ) {
```
Comment (arodic) [inline examples/js/controls/PointerLockControls.js]: I noticed that getDirection function is not used anywhere. Perhaps it is safe to remove?

**Comment by WestLangley** [comment]
I think #16171 is a cleaner implementation.

File: examples/js/controls/PointerLockControls.js (hunk 1 above)
Comment (WestLangley) [reply examples/js/controls/PointerLockControls.js]: Other users may use it.

**Comment by looeee** [comment]
I don't have any opinion on which implementation is better, but I do like the idea of exposing `minPolarAngle` and `maxPolarAngle`, and adding `change` events - whichever implementation we choose should include these.

**Comment by arodic** [comment]
I'm gonna close this PR in hopes to prevent the stall on #16171. Once it is merged, I will open another PR with the rest of the fixes.

**Pull request closed without merging**